shared/eventstore/src/main/kotlin/company/thewebhook/eventstore/EventStore.kt

```diff
@@ -0,0 +1,89 @@
+package company.thewebhook.eventstore
+
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.*
+import org.apache.kafka.clients.producer.KafkaProducer as KProducer
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.clients.producer.RecordMetadata
+import org.apache.kafka.common.errors.RecordTooLargeException
+import org.apache.kafka.common.serialization.StringSerializer
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+interface Producer<T> {
+    suspend fun connect(config: Map<String, String>)
+    suspend fun publish(topic: String, message: T): Boolean
```

Comment:
Shouldn't this be exposing some interface to handle partitions?

Reply:
Why?

Reply:
okay let's start with, what's an event store?

Reply:
It can be called "message store". Just a storage vehicle you can produce to and consume from. Can be kafka, rabbitmq, nsq etc.

Reply:
Right then the concept of partitions to be exposed becomes necessary no? Because we were discussing right, to have partitions per domain name etc.

Reply:
Partitions per domain name - we possibly may/may not do that. For ingesting webhook reqs, I am currently not doing that.

Think about it this way - if you share a partition with another domain name that produces too many webhooks than yourself, you will be rate limited. Choosing partitions on random will therefore 'normalise' it to an extent.

Reply:
But if we do require partitioning logic for some other service and not the ingestor, I say we worry about it then and not now. I'm open to the idea of exposing partitioning logic, but just that when we require it.

Reply:
Hmm Okay